### PR TITLE
docs: answer frequently asked questions

### PR DESCRIPTION
**Pull request by stevenjoezhang**
Issue resolved: https://github.com/hexojs/hexo-deployer-git/issues/71
Issue resolved: https://github.com/hexojs/hexo-deployer-git/issues/87
Issue resolved: https://github.com/hexojs/hexo-deployer-git/issues/175
Issue resolved: https://github.com/hexojs/hexo/issues/3513

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,22 @@ Additional guides:
...
 
+## Custom Domain Issue with CNAME File
+
+If you are using a custom domain with GitHub Pages, you might have encountered an issue where your custom domain configuration gets lost after deploying updates to your site. This problem occurs due to the deletion of the `CNAME` file in the root of the deployed directory during the git commit and push process performed by the `hexo-deployer-git` plugin.
+
+GitHub requires the `CNAME` file to be present in your repository to associate your custom domain with your GitHub Pages site. When this file is missing, GitHub resets the custom domain setting, leading to the site being accessible only through the default `github.io` domain.
+
+### Solution
+
+To ensure your custom domain remains active with each deployment, you need to include the `CNAME` file in your blog's source files. Here's how you can do it:
```
Comment (uiolee) [inline README.md:124]:
```suggestion
To ensure your custom domain remains active with each deployment, you need to include the `CNAME` file in your blog's `source/` directory. Here's how you can do it:
```

File: README.md [hunk 2]
```diff
@@ -113,6 +113,22 @@ Additional guides:
...
+
+1. **Create a `CNAME` File:** In the root of your blog's source directory, create a file named `CNAME`. Inside this file, enter your custom domain name. For example, if your custom domain is `example.com`, the content of the `CNAME` file should be:
```
Comment (uiolee) [inline README.md:126]:
```suggestion
1. **Create a `CNAME` File:** In the root of your blog's source directory, create a file named `CNAME`. Inside this file, enter your custom domain name. For example, if your custom domain is `example.com`, the content of the `CNAME` file should be:
```
```suggestion
1. **Create a `CNAME` File:** In the root of your blog's `source/` directory, create a file named `CNAME`. Inside this file, enter your custom domain name. For example, if your custom domain is `example.com`, the content of the `CNAME` file should be:
```

File: README.md [hunk 3]
```diff
@@ -113,6 +113,22 @@ Additional guides:
 - Create a GitHub Personal Access Token. [[Link]](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line)
 - Add authentication token to Travis CI. [[Link]](https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings)
 
+## Custom Domain Issue with CNAME File
+
+If you are using a custom domain with GitHub Pages, you might have encountered an issue where your custom domain configuration gets lost after deploying updates to your site. This problem occurs due to the deletion of the `CNAME` file in the root of the deployed directory during the git commit and push process performed by the `hexo-deployer-git` plugin.
```
Comment (uiolee) [inline README.md:118]: I think `CNAME` file was overwritten rather than deleted

**Review by uiolee — COMMENTED** [review]
Some suggestions are for reference only.

BTW, I think we had better announce to users that if they use `hexo-deployer-git`, then they should no longer make manual commits to the target repository other than using `hexo-deployer-git`. Because the changes they make will always be overwritten by `hexo-deployer-git`

**Comment by stevenjoezhang** [comment]
Thank you for your suggestion. I have added a section in the README explaining the force push behavior of this plugin.

**Review by uiolee — APPROVED** [review]

**Pull request merged**